var/spack/repos/builtin/packages/perl-math-symbolic/package.py

```diff
@@ -12,6 +12,9 @@ class PerlMathSymbolic(PerlPackage):
     homepage = "https://metacpan.org/pod/Math::Symbolic"
     url = "https://cpan.metacpan.org/authors/id/S/SM/SMUELLER/Math-Symbolic-0.612.tar.gz"
 
+    license("GPL-1.0-or-later OR Artistic-1.0-Perl")
+
     version("0.612", sha256="a9af979956c4c28683c535b5e5da3cde198c0cac2a11b3c9a129da218b3b9c08")
 
     depends_on("perl-module-build", type="build")
+    depends_on("perl-parse-recdescent")
```

Comment:
I am able to build this package for the listed version successfully without this dependency.  I can also run the new stand-alone tests (#43044).  How was this dependency determined?

Reply:
META.yml and the Makefile.PL clearly state this as a dependency, so this should be there.
A simple 'use Some-Math::Symbolic' will probably succeed, since the dependency is only 'require'd at runtime.
This was broken so far.

Reply:
In that case, why are `Memoize` (@1.01), `Data::Dumper`, and `Test::More` not included as well? 

```
$ cat Makefile.PL
# Note: this file was auto-generated by Module::Build::Compat version 0.4003
use ExtUtils::MakeMaker;
WriteMakefile
(
  'NAME' => 'Math::Symbolic',
  'VERSION_FROM' => 'lib/Math/Symbolic.pm',
  'PREREQ_PM' => {
                   'Data::Dumper' => '0',
                   'Memoize' => '1.01',
                   'Parse::RecDescent' => '0',
                   'Test::More' => 0
                 },
  'INSTALLDIRS' => 'site',
  'EXE_FILES' => [],
  'PL_FILES' => {}
)
;
```

And how can you distinguish build from run dependencies?

Reply:
@tldahlgren I had previously made a package.py for Memoize, however it is present on the corelist so was reverted: #42097
And for perl-data-dumper it is also on the corelist:
```
Data for 2019-05-22
Data::Dumper was first released with perl 5.005
```

but this package is already included in other existing spack recipes and some other config/tests, which I was going to raise in my other PR here: #41916, as to whether I should make the suggested changes or remove the package where found.
It is a similar case for perl-test-more

Reply:
Nvm.  Looks like `META.json` has that information.

Which means the following also need to be added as `runtime` dependencies:

- `Data::Dumper`
- `Memoize`

And the following as `build`-time dependency:

- `Test::More`

Do you want to take care of that in this PR?

Reply:
🤦‍♀️ I forgot to check for them being on the core.  Thanks.